claims/verifier.go

```diff
@@ -0,0 +1,9 @@
+package claims
+
+import "context"
+
+type Verifier[T any] interface {
+	// Verify will parse and verify provided token, if `AllowedAudiences` was configured those will be validated as well.
+	// Additional claims will be
+	VerifyToken(ctx context.Context, token string) (TokenClaims, T, error)
+}
```

Comment:
Or we could do something more like the [upstream implementation](https://github.com/grafana/authlib/blob/main/authn/verifier.go#L61)
```suggestion
type Verifier interface {
	// Verify will parse and verify provided token, if `AllowedAudiences` was configured those will be validated as well.
	VerifyToken(ctx context.Context, token string, claims any) (TokenClaims, error)
}
```